### ISSUE TITLE
Support 4 bit color depth

### ISSUE THREAD

**Issue by gagern** [liangzeng/node-png#5]
Reading a PNG image created from a PDF file via `convert`, which in turn calls `gs`, I get this error:

    Error: Unsupported bit depth 4

Apparently, 4 bit depth is the maximum the `pngalpha` driver for GhostScript supports, and is the default ImageMagick uses. It would be nice if these files could be read without hassle.

**Comment by liangzeng** [comment]
@gagern  I find it  https://github.com/darkskyapp/pngparse
please test it `parse` function. if ok i will integration into `node-png`

**Comment by gagern** [comment]
Yes, `pngparse` does work, it does [support](https://github.com/darkskyapp/pngparse/blob/master/index.js#L496-518) 1, 2, 4 and 8 bits per pixel. It seems somewhat slower than `node-png`, though.

**Comment by liangzeng** [comment]
@gagern I'll look. add you into collaborators.

**Comment by lukeapage** [comment]
There are a number of fragmented writers with different support. I've been refactoring pngjs to have support for any bit size. 
https://github.com/lukeapage/pngjs2

@liangzeng I wanted ownership since I was making significant changes. I also wanted the name pngjs2 since it is currently the most popular library and this allows it to be found easier in npm.
If you (or anyone else) is interested I can make a pngjs organisation and add you to collaborators.
